food-truck-frontend/src/pages/account.tsx

```diff
@@ -21,86 +19,101 @@ class AccountPageComponent extends React.Component<PageProps, PageState> {
     constructor(props: PageProps) {
         super(props);
         this.state = {
-            done: false,
             editMode: false,
             error: null,
-            username: "",
-            email: "",
-            id: 0,
             submitResultText: null,
+            user: null
         };
     }
 
-    componentDidMount() {
-        this.refreshUserInfo();
+    async componentDidMount() {
+        await this.refreshUserInfo();
     }
 
     render() {
-        if (!this.state.done) {
-            return <div/>
-        } else if (this.state.error) {
+        if (this.state.error) {
             return <h2>Error: {this.state.error}</h2>
+        }
+        else if (this.state.user == null) {
+            return <p>Please wait...</p>
         } else {
             let editForm = null;
             if (this.state.editMode) {
                 const resultP = (this.state.submitResultText !== null) ? <p>{this.state.submitResultText}</p> : null;
                 editForm = <div>
                     <Form submitUrl="/user" submitMethod="PUT" onSuccessfulSubmit={this.onSubmitSuccess}>
-                        <TextField required label="Current Password" type="password" name="password"/>
+                        <TextField  required label="Current Password" type="password" name="password"/>
                         <TextField label="New Password" type="password" name="newPassword"/>
-                        <TextField label="New Email" name="newEmail"/>
+                        <TextField label="New Email" name="newEmail" defaultValue={this.state.user.email}/>
+                        <FormControl>
+                            <Select label="New Privacy Setting" name="newPrivacySetting" defaultValue={this.state.user.privacySetting}>
+                                <MenuItem value="PUBLIC">Public</MenuItem>
+                                <MenuItem value="USERS_ONLY">Users Only</MenuItem>
+                                <MenuItem value="PRIVATE">Private</MenuItem>
+                            </Select>
+                            <FormHelperText>New Privacy Setting</FormHelperText>
+                        </FormControl>
+                        <Grid container direction="row">
+                            <p>Owner account?</p>
+                            <Switch name="newOwnerStatus" checked={this.state.user.owner}/>
```

Comment:
So we're letting them switch accounts user -> owner now? Do we block switching owner -> user? That could cause... problems.